src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static double SIZE_OF_INCREASE = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[CAPACITY];
+    }
+
+    private void increaseCapacity() {
+        if (values.length == size) {
+            int newCapacity = (int)(values.length * SIZE_OF_INCREASE);
+            values = Arrays.copyOf(values, newCapacity);
+        }
+    }
+
+    private void correctIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        increaseCapacity();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        increaseCapacity();
+        for (int i = size; i > index; i--) {
+            values[i] = values[i - 1];
+        }
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        correctIndex(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        correctIndex(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        increaseCapacity();
```

Comment:
There is no need to call 'increaseCapacity' in the 'remove' method since you are reducing the size of the array, not increasing it.